src/Symfony/Component/Scheduler/Generator/MessageGenerator.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler\Generator;
+
+use Psr\Clock\ClockInterface;
+use Symfony\Component\Clock\Clock;
+use Symfony\Component\Scheduler\Schedule;
+use Symfony\Component\Scheduler\ScheduleableInterface;
+use Symfony\Component\Scheduler\Trigger\TriggerInterface;
+
+/**
+ * @experimental
+ */
+final class MessageGenerator implements MessageGeneratorInterface
+{
+    private TriggerHeap $triggerHeap;
+    private ?\DateTimeImmutable $waitUntil;
+    private CheckpointInterface $checkpoint;
+    private Schedule $schedule;
+
+    public function __construct(
+        private readonly string $name,
+        ScheduleableInterface $scheduleProvider,
+        private readonly ClockInterface $clock = new Clock(),
+    ) {
+        $this->waitUntil = new \DateTimeImmutable('@0');
+        $this->schedule = $scheduleProvider->getSchedule();
+        $this->checkpoint = new Checkpoint('scheduler_checkpoint_'.$name, $this->schedule->getLock(), $this->schedule->getState());
+    }
+
+    public function withCheckpoint(CheckpointInterface $checkpoint): static
+    {
+        $generator = clone $this;
+        $generator->checkpoint = $checkpoint;
+
+        return $generator;
+    }
+
+    public function getMessages(): \Generator
+    {
+        if (!$this->waitUntil ||
```

Comment:
we usually put boolean operator on the start of the next line

Reply:
sure